cognite/neat/_graph/transformers/_prune_graph.py

```diff
@@ -1,84 +1,131 @@
+        predicate_property_name = "Name", delete_connectiong_node=True):
 
-        Graph after flattening nodes with destination_node_type = rdf:type(TextObject), property_predicate = :value,
-        and property_name = "value":
+        <http://purl.org/cognite/neat/Pump-1> a ns1:Pump ;
+        ns1:ItemTagAssignmentClass "D-20PSV0002" .
 
-        node(A, rdf:type(Pump)) -(predicate("vendor"))> Literal("CompanyX")
+        Number of triples after operation: 5 - 1*3 = 2
 
     Args:
         destination_node_type: RDF.type of edge Node
-        property_predicate: Predicate to use when resolving the value from the edge node
-        property_name: name of the property that the intermediate node is pointing to
+        namespace: RDF Namespace to use when querying the graph
+        value_property_name: str with name of the property that holds the value attached to the intermediate node
+        predicate_property_name: Optional str of the property name that holds the new predicate to use when resolving
+        the intermediate connection.
         delete_connecting_node: bool if the intermediate Node and Edge between source Node
-                                and target property should be deleted. Defaults to True.
+                                and target property should be deleted. Defaults to False.
     """
 
     description: str = "Prunes the graph of specified node types that do not have connections to other nodes."
-    _query_template: str = """SELECT ?sourceNode ?property ?destinationNode ?value WHERE {{
-                                     ?sourceNode ?property ?destinationNode .
-                                     ?destinationNode a <{destination_node_type}> .
-                                     ?destinationNode <{property_predicate}> ?{property_name} . }}"""
+
+    _query_template_keep_old_predicate: str = """
+    SELECT ?sourceNode ?property_source ?destinationNode ?property_destination ?property_value WHERE {{
+        ?sourceNode ?property_source ?destinationNode .
+        ?destinationNode ?property_destination ?property_value .
+        ?destinationNode a <{destination_node_type}> .
+        ?destinationNode <{value_property_predicate}> ?property_value . }}"""
+
+    _query_template_new_predicate: str = """
+    SELECT ?sourceNode ?property ?destinationNode ?predicate_value ?property_value WHERE {{
+        ?sourceNode ?property ?destinationNode .
+        ?destinationNode a <{destination_node_type}> .
+        ?destinationNode <{predicate_property_predicate}> ?predicate_value .
+        ?destinationNode <{value_property_predicate}> ?property_value . }}"""
 
     def __init__(
         self,
         destination_node_type: URIRef,
-        property_predicate: Namespace,
-        property_name: str,
-        delete_connecting_node: bool = True,
+        namespace: Namespace,
+        value_property_name: str,
+        delete_connecting_node: bool = False,
+        predicate_property_name: str | None = None,
```

Comment:
rename to `property_name_holding_new_property` maybe ? 

lift this arg above `delete_connecting_node`